File: LibraryTests/Java/TestAssignment.java
import java.io.File;
import java.io.PrintWriter;
import java.math.BigInteger;
import java.util.Random;

class TestAssignment {
    public static void main(String[] args) {
        // Declaration and initialization
        Long startingValue = Long.parseLong(args[0]);
        Long upperBound = Long.parseLong(args[1]);
        Long step = Long.parseLong(args[2]);
        PrintWriter printWriter = null;

        try {
            printWriter = new PrintWriter(new File("./JavaAssignmentResults.csv"));
            printWriter.write("Value,Time Taken to Assign Values (Java)\n");
            printWriter.flush();
        } catch (Exception e) {
            e.printStackTrace();
            return;
        }

        Random random = new Random();
        Character[] numbers = new Character[]{'0','1','2','3','4','5','6','7','8','9'};

        for (Long trial = startingValue; trial <= upperBound; trial += step) {

            String str = new String();
            for(long i = 0 ; i < trial; i++)
            {
                str += numbers[random.nextInt(numbers.length)];
            }

            // Actual calculation
            long start = System.nanoTime();
            BigInteger bigInteger = new BigInteger(str);
            long end = System.nanoTime();

            // Write the result to the file
            printWriter.println(trial + "," + (end - start) / 1000);
            printWriter.flush();
            // Showing the result
            //System.out.println(trial + "! = " + bigInteger.toString());
            System.out.println("\nAssignment of a random " + trial + " characters long number took " + (end - start) / 1000 + " microSeconds");
        }

        printWriter.close();
        System.out.println("Simulation Ended");

    }
}

File: LibraryTests/C/test_assignment.c
#include <stdio.h>
#include <stdlib.h>
#include <gmp.h>
#include <time.h>

// A very simple program to run factorial on large numbers with arbitrary precision
int main(int argc, char const *argv[])
{
    // Variables
    long startingValue, endingValue, step;
    mpz_t n;
    clock_t start, end;
    double cpuTimeUsed;
    srand(time(NULL));

    // Parameters
    printf("Insert the starting value: ");
    scanf("%ld", &startingValue);
    printf("Insert the ending value: ");
    scanf("%ld", &endingValue);
    printf("Insert the step: ");
    scanf("%ld", &step);

    // Initialize the file writer
    FILE *file = fopen("./CAssignmentResults.csv", "w");
    fprintf(file, "Value,Time Taken to Assign Values (C)\n");
    fflush(file);

    // Calculating the specified number of times
    for (long trial = startingValue; trial <= endingValue; trial += step)
    {
        // Building the string
        char *str = malloc(trial * sizeof(char));
        for (long i = 0; i < trial; i++)
        {
            str[i] = (char)(rand() % 10 + '0');
        }

        start = clock();

        // Initialization And set to a given String
        mpz_init_set_str(n, str, 10);

        end = clock();

        cpuTimeUsed = ((double)(end - start)) / CLOCKS_PER_SEC;
        long timeInMicroiseconds = cpuTimeUsed * 1000000;

        // Printing the result
        printf("\nTime taken by %ld long input = %ld microseconds\n", trial, timeInMicroiseconds);
        fprintf(file, "%ld,%ld\n", trial, timeInMicroiseconds);
        fflush(file);
    }

    fclose(file);
    printf("The test is over, check out the results in ./CAssignmentResults\n");

    return 0;
}
